x-pack/plugins/ml/public/application/components/import_export_jobs/export_jobs_flyout/export_jobs_flyout.tsx

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, useState, useEffect } from 'react';
+import { FormattedMessage } from '@kbn/i18n/react';
+import {
+  EuiFlyout,
+  EuiFlyoutFooter,
+  EuiFlyoutHeader,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButtonEmpty,
+  EuiButton,
+  EuiFlyoutBody,
+  EuiTitle,
+  EuiSpacer,
+  EuiCheckbox,
+  EuiTabs,
+  EuiTab,
+} from '@elastic/eui';
+
+import { useMlApiContext } from '../../../contexts/kibana';
+import { utilsProvider } from './utils';
+import type { JobType } from '../../../../../common/types/saved_objects';
+
+interface Props {
+  isDisabled: boolean;
+  currentTab: JobType;
+}
+
+export const ExportJobsFlyout: FC<Props> = ({ isDisabled, currentTab }) => {
+  const mlApiServices = useMlApiContext();
+  const {
+    getJobs,
+    dataFrameAnalytics: { getDataFrameAnalytics },
+  } = mlApiServices;
+
+  const { exportAnomalyDetectionJobs, exportDataframeAnalyticsJobs } = utilsProvider(mlApiServices);
+
+  const [showFlyout, setShowFlyout] = useState(false);
+  const [adJobIds, setAdJobIds] = useState<string[]>([]);
+  const [dfaJobIds, setDfaJobIds] = useState<string[]>([]);
+  const [selectedJobIds, setSelectedJobIds] = useState<string[]>([]);
+  const [exporting, setExporting] = useState(false);
+  const [selectedJobType, setSelectedJobType] = useState<JobType>(currentTab);
+
+  useEffect(() => {
+    setAdJobIds([]);
+    setSelectedJobIds([]);
+    setExporting(false);
+    setSelectedJobType(currentTab);
+
+    if (showFlyout) {
+      getJobs().then(({ jobs }) => {
+        setAdJobIds(jobs.map((j) => j.job_id));
+      });
+      getDataFrameAnalytics().then(({ data_frame_analytics: dataFrameAnalytics }) => {
+        setDfaJobIds(dataFrameAnalytics.map((j) => j.id));
+      });
+    }
+  }, [showFlyout]);
+
+  function toggleFlyout() {
+    setShowFlyout(!showFlyout);
+  }
+
+  async function onExport() {
+    setExporting(true);
```

Comment:
Is it worth showing a toast when the export begins, as happens in the Kibana saved objects export, to provide extra indication that something is happening?

![image](https://user-images.githubusercontent.com/7405507/126146604-61a02f6c-7936-4971-9d82-7986fae8e313.png)

Reply:
added this toast here https://github.com/elastic/kibana/pull/101037/commits/a5efc164791f4c94d4846785cb7da67eee399ae3